### PR TITLE
Specific classes for geocoding response

### PR DESCRIPTION
**Pull request by pawel-wroniszewski**
This may be a breaking change for clients sending requests to geocoding endpoint. 

Maybe it can be made non-breaking if the GeocodingResponse extends FeatureCollection from geojson 3rd party lib and is brought as a Jackson mix-in https://medium.com/@shankar.ganesh.1234/jackson-mixin-a-simple-guide-to-a-powerful-feature-d984341dc9e2

AFAIK we would need to overload the FeatureCollection constructor. Let me know if it makes sense to explore it further, I have a feeling that it wouldn't be too clean. Maybe we can simply bump the major version of the sdk and we're good.

### CODE REVIEW

**Review by gergelytraveltime — APPROVED** [review]

**Review by mjanuszkiewicz-tt — APPROVED** [review]

**Comment by danielnaumau** [comment]
I think that it would be better to make your own implementation of geoJson generic and replace it in another requests. In that case we'll be able to specify all properties everywhere. Now it looks too messy: geocoding uses our own version of geoJson and another requests use the 3rd party library.

**Review by danielnaumau — CHANGES_REQUESTED** [review]
I think that it would be better to make your own implementation of geoJson generic and replace it in another requests. In that case we'll be able to specify all properties everywhere. Now it looks too messy: geocoding uses our own version of geoJson and another requests use the 3rd party library.

**Comment by pawel-wroniszewski** [comment]
@danielnaumau I looked into this and I'm not fully convinced we should reimplement the geojson for TimeMap. In case of geocoding the custom classes serve a very particular purpose - they bring the important stuff, which are our custom properties,  to the foreground. Our response format is aligned with geojson, but there's not much of geometry being returned (only a point).
 
In case of TimeMap, the geometry is what the client wants and there is not much of custom properties (there's nothing in the docs, but I see that "is_only_walking" property is being returned sometimes). I think the geojson library is a natural choice, reimplementing it will not bring any added value, and I believe it may cause unnecessary headache.

**Comment by pawel-wroniszewski** [comment]
@danielnaumau anyway I'll try to implement the MultiPolygon support and see how difficult it is.

**Comment by danielnaumau** [comment]
> @danielnaumau anyway I'll try to implement the MultiPolygon support and see how difficult it is.

https://github.com/traveltime-dev/traveltime-utils/blob/master/modules/geo-json/src/main/scala/com/traveltime/GeoJson.scala

https://github.com/traveltime-dev/traveltime-utils/blob/master/modules/geo-json/src/main/scala/com/traveltime/GeoJsonCodecs.scala

I implemented it in Scala so it's not very difficult.

**Comment by pawel-wroniszewski** [comment]
Thanks!

**Review by PKJonas — APPROVED** [review]

File: src/main/java/com/traveltime/sdk/dto/responses/geocoding/Properties.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    String city;
+    String town;
+    String district;
+    String country;
+    String countryCode;
+    String localAdmin;
+    String continent;
+    String postcode;
+
+    com.traveltime.sdk.dto.responses.mapinfo.Feature features;
```
Comment (danielnaumau) [inline src/main/java/com/traveltime/sdk/dto/responses/geocoding/Properties.java:44]: should it be here?

File: src/main/java/com/traveltime/sdk/dto/responses/geojson/Coordinates.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.traveltime.sdk.dto.responses.geojson;
```
Comment (danielnaumau) [inline src/main/java/com/traveltime/sdk/dto/responses/geojson/Coordinates.java:1]: I think it's better to take it out from responses(com.traveltime.sdk.geojson or smith like that)

**Review by danielnaumau — APPROVED** [review]

File: src/main/java/com/traveltime/sdk/dto/responses/geocoding/Properties.java (hunk 1 above)
Comment (pawel-wroniszewski) [reply src/main/java/com/traveltime/sdk/dto/responses/geocoding/Properties.java:44]: The 'features' field of type mapinfo.Feature is part of the geocoding properties, however I removed the fully qualified path as it was not needed anymore.

File: src/main/java/com/traveltime/sdk/dto/responses/geojson/Coordinates.java (hunk 2 above)
Comment (pawel-wroniszewski) [reply src/main/java/com/traveltime/sdk/dto/responses/geojson/Coordinates.java:1]: Do you mean to reuse the com.traveltime.sdk.dto.common.Coordinates class? 
I was thinking about it, but I didn't see an easy way to serialize/deserialize a single class in two different way. I think we would need to introduce two separate methods in jsonUtils - to/from json + to/from geoJson  with two different Mappers. Is it worth it?

Comment (pawel-wroniszewski) [reply src/main/java/com/traveltime/sdk/dto/responses/geojson/Coordinates.java]: Ok, I'll change the package

**Review by mjanuszkiewicz-tt — APPROVED** [review]

**Pull request merged**